cli/cliui/select.go

```diff
@@ -114,35 +253,191 @@ func MultiSelect(inv *serpent.Invocation, opts MultiSelectOptions) ([]string, er
 		return opts.Defaults, nil
 	}
 
-	prompt := &survey.MultiSelect{
-		Options: opts.Options,
-		Default: opts.Defaults,
-		Message: opts.Message,
+	options := make([]*multiSelectOption, len(opts.Options))
+	for i, option := range opts.Options {
+		chosen := false
+		for _, d := range opts.Defaults {
+			if option == d {
+				chosen = true
+			}
+		}
+
+		options[i] = &multiSelectOption{
+			option: option,
+			chosen: chosen,
+		}
 	}
 
-	var values []string
-	err := survey.AskOne(prompt, &values, survey.WithStdio(fileReadWriter{
-		Reader: inv.Stdin,
-	}, fileReadWriter{
-		Writer: inv.Stdout,
-	}, inv.Stdout))
-	if errors.Is(err, terminal.InterruptErr) {
-		return nil, Canceled
+	initialModel := multiSelectModel{
+		search:  textinput.New(),
+		options: options,
+		message: opts.Message,
+	}
+
+	initialModel.search.Prompt = ""
+	initialModel.search.Focus()
+
+	m, err := tea.NewProgram(
+		initialModel,
+		tea.WithInput(inv.Stdin),
+		tea.WithOutput(inv.Stdout),
+	).Run()
+
+	values := []string{}
+	if m, ok := m.(multiSelectModel); ok {
```

Comment:
What if `!ok`?

Reply:
`ok` _should_ never be false here. Happy to return an error if that were to happen though.

Reply:
Have changed this to error now if `!ok`